script.js

```diff
@@ -1,10 +1,13 @@
 function shifumi() {
     const user_choice = ['rock', 'paper', 'scissors']
-    const choice = prompt(`Choose one of the following : ${user_choice.join(', ')}.`)
+    let choice = prompt(`Choose one of the following : ${user_choice.join(', ')}.`)
+    if (!user_choice.includes(choice)) {
+        alert(`Attention, ${choice} is not one of the ${user_choice.length} choices`);
+        shifumi()
```

Comment:
you will end up with a problem here. The function will be re-executed, while the current one will continue to be executed. The early return was actually the good move. 

And if you add a return here, you can also remove something... :)

Reply:
I guess this is the final version now :)
function shifumi() {
    const user_choice = ['rock', 'paper', 'scissors']
    let choice = prompt(`Choose one of the following : ${user_choice.join(', ')}.`)
    if (!user_choice.includes(choice)) {
        alert(`Attention, ${choice} is not one of the ${user_choice.length} choices`);
        return shifumi()
    }
    if (user_choice.includes(choice)) {
        alert(`Congratulations, you chose ${choice}`);
    }
}
shifumi();

Reply:
Better, but still too much code, you can delete a condition ;)

Reply:
Testing would help